File: grida-core/src/main/java/org/grida/datetime/DateTimeUtil.java
package org.grida.datetime;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.Date;

public class DateTimeUtil {

    private DateTimeUtil() {
    }

    public static Date toDate(LocalDateTime localDateTime) {
        return Date.from(localDateTime.atZone(ZoneId.systemDefault()).toInstant());
    }
}

File: grida-common/src/main/kotlin/org/grida/datetime/DateTimeUtil.kt
package org.grida.datetime

import java.time.LocalDate
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object DateTimeUtil {

    fun toDefaultDateTimeFormat(localDateTime: LocalDateTime): String {
        return localDateTime.format(DateTimeFormatter.ofPattern(DEFAULT_DATE_TIME_FORMAT))
    }

    fun toDefaultDateFormat(localDate: LocalDate): String {
        return localDate.format(DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT))
    }

    fun toLocalDateTime(rawLocalDateTime: String): LocalDateTime {
        return LocalDateTime.parse(rawLocalDateTime, DateTimeFormatter.ofPattern(DEFAULT_DATE_TIME_FORMAT))
    }

    fun toLocalDate(rawLocalDate: String): LocalDate {
        return LocalDate.parse(rawLocalDate, DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT))
    }
}
